docsite/source/error-handling.html.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Error Handling
+layout: gem-single
+name: dry-operation
+---
+
+When using dry-operation, errors are handled through the `Failure` type from [dry-monads](/gems/dry-monads/). Each step in your operation should return either a `Success` or `Failure` result. When a step returns a `Failure`, the operation short-circuits, skipping the remaining steps and returning the failure immediately.
+
+You'll usually handle the failure from the call site, where you can pattern match on the result to handle success and failure cases. However, sometimes it's useful to encapsulate some error handling logic within the operation itself.
+
+### Global error handling
+
+You can define a global failure handler by implementing an `#on_failure` method in your operation class:
```

Comment:
We have an example of an on_failure below, but it's not really clear if this method has particular responsibilities that it must fulfil. For example, does it matter what this method returns? Does the return value of on_failure become the return value of the overall operation call? I'm pretty sure the answer is no to this, that `on_failure` is only really used for side effects only. But it would be good to clarify this :)

Reply:
Good point, added!